velox/common/compression/v2/Lz4Compression.cpp

```diff
@@ -0,0 +1,529 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <glog/logging.h>
+
+#include "velox/common/base/Exceptions.h"
+#include "velox/common/compression/v2/Lz4Compression.h"
+
+namespace facebook::velox::common {
+
+namespace {
+
+void lz4Error(LZ4F_errorCode_t errorCode, const char* prefixMessage) {
+  VELOX_FAIL(prefixMessage, LZ4F_getErrorName(errorCode));
+}
+
+LZ4F_preferences_t defaultPreferences() {
+  LZ4F_preferences_t prefs;
+  memset(&prefs, 0, sizeof(prefs));
+  return prefs;
+}
+
+LZ4F_preferences_t defaultPreferences(int compressionLevel) {
+  LZ4F_preferences_t prefs = defaultPreferences();
+  prefs.compressionLevel = compressionLevel;
+  return prefs;
+}
+} // namespace
+
+class LZ4Compressor : public Compressor {
+ public:
+  explicit LZ4Compressor(int32_t compressionLevel);
+
+  ~LZ4Compressor() override;
+
+  void init();
+
+  CompressResult compress(
+      uint64_t inputLength,
+      const uint8_t* input,
+      uint64_t outputLength,
+      uint8_t* output) override;
+
+  FlushResult flush(uint64_t outputLength, uint8_t* output) override;
+
+  EndResult end(uint64_t outputLength, uint8_t* output) override;
+
+ protected:
+  void
+  compressBegin(uint8_t* output, size_t& outputLen, uint64_t& bytesWritten);
```

Comment:
Put `void` and `compressBegin` to the same line?

Reply:
This line is auto-formatted.